src/librbd/api/Group.cc

```diff
@@ -316,6 +317,22 @@ int Group<I>::image_remove_by_id(librados::IoCtx& group_ioctx,
     return r;
   }
 
+  mirror::PromotionState promotion_state;
+  cls::rbd::MirrorGroup mirror_group;
+  C_SaferCond ctx;
+  auto request = mirror::GroupGetInfoRequest<I>::create(
+    group_ioctx, "", group_id, &mirror_group, &promotion_state, &ctx);
+  request->send();
+  r = ctx.wait();
+  if (r < 0 && r != -ENOENT && r != -EOPNOTSUPP) {
+    lderr(cct) << "failed to get mirror group info: "
+               << cpp_strerror(r) << dendl;
+    return r;
+  } else if (r == 0 && promotion_state != mirror::PROMOTION_STATE_PRIMARY) {
+    lderr(cct) << "cannot remove image from seconday group" << dendl;
```

Comment:
```suggestion
    lderr(cct) << "cannot remove image from non-primary group" << dendl;
```

Because we are checking for `!PROMOTION_STATE_PRIMARY` only the group state can be anything other than it (i.e. PROMOTION_STATE_UNKNOWN/PROMOTION_STATE_NON_PRIMARY/PROMOTION_STATE_ORPHAN). We are not quite sure if this is secondary group just by above condition.